### ISSUE TITLE
Reason for nw.js being pinned at 0.12.1?

### ISSUE THREAD

**Issue by mathieumg** [plaidchat/plaidchat#97]
Is there any reason why it's not a more relaxed semver or simply 0.12.2? Using my home machine (Ubuntu 12.04 x64), I get the following crash when I try to start Plaidchat:

```bash
[11509:0628/212645:ERROR:browser_main_loop.cc(170)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
LaunchProcess: failed to execvp:
/home/mathieu/local/lib/node_modules/plaidchat/node_modules/nw/nwjs/nw
LaunchProcess: failed to execvp:
/proc/self/exe
[11509:0628/212645:ERROR:child_process_launcher.cc(479)] Failed to launch child process
[11509:0628/212645:ERROR:channel.cc(305)] RawChannel read error (connection broken)
[11509:0628/212645:ERROR:child_process_launcher.cc(479)] Failed to launch child process
```

The problem is fixed when bumping nw.js to 0.12.2 See https://github.com/nwjs/nw.js/issues/2669

If there's no valid reason, I don't mind submitting a PR.

**Comment by twolfson** [comment]
Previously, it wasn't pinned at all and attempting to install 0.13 when it was unavailable. I think we are open to upgrading as new releases come out. Feel free to send over that PR and I will approve/release =)

**Comment by mathieumg** [comment]
> it wasn't pinned at all and attempting to install 0.13

What semvar did it have? Unless it had `latest` or `0.x`, even `^0.12.0` wouldn't upgrade to `0.13`.

**Comment by twolfson** [comment]
This was before we used the `npm` installed flavor of `nw.js`. We were on `grunt-node-webkit-builder` which by default downloads the `latest` from `nw.js` regardless of it being stable/not:

https://github.com/mllrsohn/grunt-node-webkit-builder

https://github.com/mllrsohn/node-webkit-builder#optionsversion

**Comment by mathieumg** [comment]
I see, thanks for the explanation! So if I set it to `^0.12.2` it's fine by you?

**Comment by twolfson** [comment]
We could probably loosen our semver instead of using an exact pin. I have noticed a bit of instability between `nw.js's` minor releases (e.g. `0.10` to `0.12` has a lot of differences). I would prob choose `~0.12.1`.

**Comment by mathieumg** [comment]
Below 1.0, these are actually considered two major ("breaking") releases under semver.

**Comment by twolfson** [comment]
Yea, semver is weird. There is a lot of frustration about that inconsistency since it breaks `~`/`^` from being consistent before/after `1.0.0`.

To provide an example of a breaking change in the `0.10`/`0.12`:https://github.com/nwjs/nw.js/issues/3483

Let's go with `~0.12.2` to prevent rushed releases =)

**Comment by mathieumg** [comment]
Done in https://github.com/plaidchat/plaidchat/issues/98

**Comment by twolfson** [comment]
This has been landed in `2.5.1`. Thanks for the bug report and quick fix =)
